owncloudData/src/main/java/com/owncloud/android/data/oauth/repository/OCOAuthRepository.kt

```diff
@@ -19,18 +19,41 @@
 package com.owncloud.android.data.oauth.repository
 
 import com.owncloud.android.data.oauth.datasources.RemoteOAuthDataSource
+import com.owncloud.android.data.webfinger.datasources.WebfingerRemoteDatasource
 import com.owncloud.android.domain.authentication.oauth.OAuthRepository
 import com.owncloud.android.domain.authentication.oauth.model.ClientRegistrationInfo
 import com.owncloud.android.domain.authentication.oauth.model.ClientRegistrationRequest
 import com.owncloud.android.domain.authentication.oauth.model.OIDCServerConfiguration
 import com.owncloud.android.domain.authentication.oauth.model.TokenRequest
 import com.owncloud.android.domain.authentication.oauth.model.TokenResponse
+import com.owncloud.android.domain.exceptions.FileNotFoundException
+import com.owncloud.android.domain.webfinger.model.WebfingerRel
+import timber.log.Timber
 
 class OCOAuthRepository(
-    private val oidcRemoteOAuthDataSource: RemoteOAuthDataSource
+    private val oidcRemoteOAuthDataSource: RemoteOAuthDataSource,
+    private val webfingerRemoteDatasource: WebfingerRemoteDatasource,
```

Comment:
To keep consistency with the rest of datasources, maybe this class should be called `RemoteWebfingerDataSource` and thus this property `remoteWebfingerDataSource`, do you agree?